compiler/vm/compilerbridge.nim

```diff
@@ -152,19 +154,43 @@ proc unpackResult(res: sink ExecutionResult; config: ConfigRef, node: PNode): PN
     result = config.newError(node, errKind, rId, instLoc(-1)):
                              [newIntNode(nkIntLit, ord(stId))]
 
-proc buildError(c: var TCtx, thread: VmThread, event: sink VMReport): ExecErrorReport  =
+proc createLegacyStackTrace(
+    c: TCtx, 
+    thread: VmThread, 
+    instLoc: InstantiationInfo = instLoc(-1)
+  ): VMReport =
+  let st = createStackTrace(c, thread)
+  result = VMReport(kind: rvmStackTrace,
+                    currentExceptionA: st.currentExceptionA,
+                    currentExceptionB: st.currentExceptionB,
+                    traceReason: vmEventToLegacyReportKind(st.traceReason),
+                    stacktrace: st.stacktrace,
+                    skipped: st.skipped,
+                    location: some source(c, thread),
+                    reportInst: toReportLineInfo(instLoc))
+
+proc buildError(c: TCtx, thread: VmThread, event: sink VMReport): ExecErrorReport  =
   ## Creates an `ExecErrorReport` with the `event` and a stack-trace for
   ## `thread`
-  result.stackTrace = createStackTrace(c, thread)
+  result.stackTrace = createLegacyStackTrace(c, thread, instLoc(-1))
```

Comment:
You can change the type of `ExecErrorReport.stackTrace` to `VmStackTrace` instead and then perform the translation in `unpackResult`. That moves the transition point further upwards.

Reply:
I did that earlier but went back given the dependency on thread for the program counter, etc. I'll give it another go soon because other things are cleaner already.

Reply:
done